tap_github/tap.py

```diff
@@ -56,8 +58,31 @@ class TapGitHub(Tap):
                 "streams (such as repositories) if it is not selected but children are"
             ),
         ),
+        th.Property(
+            "exclude",
+            th.ArrayType(th.StringType),
+            description="List of streams to exclude by name.",
+        ),
     ).to_dict()
 
+    @property
+    def _singer_catalog(self) -> Catalog:
+        """Return a Catalog object.
+
+        Returns:
+            :class:`singer_sdk.helpers._singer.Catalog`.
+        """
+        excluded_streams = self.config.get("exclude", [])
+
+        def get_stream_entry(stream: Stream):
+            singer_catalog_entry = stream._singer_catalog_entry
+            singer_catalog_entry.metadata.root.selected = (
+                stream.tap_stream_id not in excluded_streams
+            )
+            return (stream.tap_stream_id, singer_catalog_entry)
```

Comment:
Should this return an instance of `CatalogEntry`?

Reply:
We could, I just re-used the minimal object that the default function uses